functions/go-mongo/main.go

```diff
@@ -0,0 +1,91 @@
+package example
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var client *mongo.Client
+
+type RandomDocument struct {
+	ID   int    `bson:"id"`
+	Name string `bson:"name"`
+}
+
+func init() {
+	mongoPublicEndpoint := os.Getenv("MONGO_PUBLIC_ENDPOINT")
+	if mongoPublicEndpoint == "" {
+		panic("MONGO_PUBLIC_ENDPOINT is required")
+	}
+
+	mongoUser := os.Getenv("MONGO_USER")
+	if mongoUser == "" {
+		panic("MONGO_USER is required")
+	}
+
+	mongoPassword := url.PathEscape(os.Getenv("MONGO_PASSWORD"))
+	if mongoPassword == "" {
+		panic("MONGO_PASSWORD is required")
+	}
+
+	// This is a basic sample that does not use certificate for authentication, not recommended for production.
+	mongo_uri := fmt.Sprintf(`mongodb+srv://%s:%s@%s/?tls=true&tlsInsecure=true`,
```

Comment:
issue: code panics if password contains a `@` (escape issue): `error parsing uri: unescaped @ sign in user info`.

I'm not sure if we can escape it, but I feel the connection should be done using `options.Credential` instead of formatting the string. I've tried with:

```go
credentials := options.Credential{
  Username: mongoUser,
  Password: mongoPassword,
}

var err error
client, err = mongo.Connect(context.TODO(), options.Client().ApplyURI(mongo_uri).SetAuth(credentials))
```

But I'm facing another issue (`unable to authenticate using "SCRAM-SHA-256"`). This might have something to do with `AuthSource` field in `Credential`, but couldn't figure out.

Anyway, like this, it doesn't work with special `@` character.